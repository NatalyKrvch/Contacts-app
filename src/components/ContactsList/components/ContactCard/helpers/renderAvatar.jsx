import { Image } from 'react-bootstrap'
import { MdAccountCircle } from 'react-icons/md'

const renderAvatar = (avatar_url) => {
  return avatar_url ? (
    <Image
      src={avatar_url}
      roundedCircle
      width={48}
      height={48}
      alt="avatar"
    />
  ) : (
    <MdAccountCircle size={48} color="#e8eaed" />
  )
}

export default renderAvatar
