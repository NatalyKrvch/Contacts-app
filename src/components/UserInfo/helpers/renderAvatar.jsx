import { Image } from 'react-bootstrap'
import { MdAccountCircle } from 'react-icons/md'

const renderAvatar = (avatarUrl, size) => {
  return avatarUrl ? (
    <Image
      src={avatarUrl}
      roundedCircle
      width={size}
      height={size}
      alt="avatar"
    />
  ) : (
    <MdAccountCircle size={size} color="#e8eaed" />
  )
}

export default renderAvatar
