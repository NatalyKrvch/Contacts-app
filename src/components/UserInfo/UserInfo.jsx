import { Card } from 'react-bootstrap'
import renderAvatar from './helpers/renderAvatar'

const UserInfo = ({ firstName, lastName, email, avatarUrl, avatarSize }) => {
  return (
    <div className="d-flex align-items-center">
      {renderAvatar(avatarUrl, avatarSize)}
      <div className="ms-3">
        <Card.Title className="contact-title">
          {firstName} {lastName}
        </Card.Title>
        <Card.Text className="contact-email">{email}</Card.Text>
      </div>
    </div>
  )
}

export default UserInfo
