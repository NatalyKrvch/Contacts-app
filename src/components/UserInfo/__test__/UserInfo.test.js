import { render, screen } from '@testing-library/react';
import UserInfo from '../UserInfo';
import renderAvatar from '../helpers/renderAvatar';

jest.mock('../helpers/renderAvatar');

describe('UserInfo', () => {
  const mockProps = {
    firstName: 'John',
    lastName: 'Doe',
    email: 'john.doe@example.com',
    avatarUrl: 'https://example.com/avatar.jpg',
    avatarSize: 50,
  };

  it('renders component correctly', () => {
    const { container } = render(<UserInfo {...mockProps} />);
    
    expect(container).toMatchSnapshot();
  });

  it('renders user info with avatar', () => {
    renderAvatar.mockReturnValue(
      <img src="https://example.com/avatar.jpg" alt="avatar" />
    );
    const { container } = render(<UserInfo {...mockProps} />);
    
    expect(screen.getByAltText('avatar')).toBeInTheDocument();
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('john.doe@example.com')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders user info with default avatar', () => {
    renderAvatar.mockReturnValue(
      <svg data-testid="default-avatar-icon"></svg>
    );
    const propsWithoutAvatar = { ...mockProps, avatarUrl: '' };
    const { container } = render(<UserInfo {...propsWithoutAvatar} />);
    
    expect(screen.getByTestId('default-avatar-icon')).toBeInTheDocument();
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('john.doe@example.com')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});

