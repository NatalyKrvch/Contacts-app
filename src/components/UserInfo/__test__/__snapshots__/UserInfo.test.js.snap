// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo renders component correctly 1`] = `
<div>
  <div
    class="d-flex align-items-center"
  >
    <div
      class="ms-3"
    >
      <div
        class="contact-title card-title h5"
      >
        John
         
        Doe
      </div>
      <p
        class="contact-email card-text"
      >
        john.doe@example.com
      </p>
    </div>
  </div>
</div>
`;

exports[`UserInfo renders user info with avatar 1`] = `
<div>
  <div
    class="d-flex align-items-center"
  >
    <img
      alt="avatar"
      src="https://example.com/avatar.jpg"
    />
    <div
      class="ms-3"
    >
      <div
        class="contact-title card-title h5"
      >
        John
         
        Doe
      </div>
      <p
        class="contact-email card-text"
      >
        john.doe@example.com
      </p>
    </div>
  </div>
</div>
`;

exports[`UserInfo renders user info with default avatar 1`] = `
<div>
  <div
    class="d-flex align-items-center"
  >
    <svg
      data-testid="default-avatar-icon"
    />
    <div
      class="ms-3"
    >
      <div
        class="contact-title card-title h5"
      >
        John
         
        Doe
      </div>
      <p
        class="contact-email card-text"
      >
        john.doe@example.com
      </p>
    </div>
  </div>
</div>
`;
